File: 14-Longest Common Pre-最长公共前缀fix/14-Longest Common Pre-最长公共前缀fix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int len = strs.length;
        StringBuilder sb = new StringBuilder();
        Arrays.sort(strs);
        int minLen = Math.min(strs[0].length(), strs[len - 1].length());
        for(int i = 0; i < minLen ; i++) {
        	if(strs[0].charAt(i) == strs[len - 1].charAt(i)) {
        		sb.append(strs[0].charAt(i));
        	}else {
        		return sb.toString();
        	}
        }
        return sb.toString();
    }
}

File: 14-Longest Common Pre-最长公共前缀fix/14-Longest Common Pre-最长公共前缀fix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let res = ''
    const sortedStrs = strs.sort()
    const aryLen = sortedStrs.length
    const minLen = Math.min(sortedStrs[0].length, sortedStrs[aryLen - 1].length)
    for(i = 0; i < minLen; i++){
        if(sortedStrs[0][i] === sortedStrs[aryLen - 1][i]){
            res += sortedStrs[0][i]
        }else{
            return res
        }
    }
    return res
};
